### PR TITLE
Documentation upgrade for new design

### PR DESCRIPTION
**Pull request by iuliadmtru**
This is a rather large PR, but most changes come from PR #48 so it is actually not that large. For the first part of the changes see PR #48. For the rest:

I started restructuring the documentation and adding documentation where it was missing. The idea is to have a clear documentation that is easy to understand not only for the users coming from R but also for Julia-only users, as suggested [here](https://discourse.julialang.org/t/suggestions-for-the-design-of-survey-jl/86381/3?u=iuliadmtru). This is only the beginning, I did not add much. This is why I will mark this as a draft instead of a PR.

### CODE REVIEW

File: src/SurveyDesign.jl [hunk 1]
```diff
@@ -1,62 +1,107 @@
...
+    printstyled("\n    popsize: "; bold = true)
+    print(design.popsize)
+    printstyled("\n    sampsize: "; bold = true)
+    print(design.sampsize)
 end
 
 """
-A `StratifiedSample` object holds information necessary for surveys sampled by
-stratification.
+    StratifiedSample <: AbstractSurveyDesign
```
Comment (ayushpatnaikgit) [inline src/SurveyDesign.jl:94]:
```suggestion
    `StratifiedSample <: AbstractSurveyDesign`
```

File: src/SurveyDesign.jl [hunk 2]
```diff
@@ -71,41 +116,57 @@ struct StratifiedSample <: AbstractSurveyDesign
...
+    printstyled("\n    popsize: "; bold = true)
+    print(design.popsize)
+    printstyled("\n    sampsize: "; bold = true)
+    print(design.sampsize)
 end
 
 """
-A `ClusterSample` object holds information necessary for surveys sampled by
-clustering.
+    ClusterSample <: AbstractSurveyDesign
```
Comment (ayushpatnaikgit) [inline src/SurveyDesign.jl:138]:
```suggestion
    `ClusterSample <: AbstractSurveyDesign`
```

Comment (iuliadmtru) [reply src/SurveyDesign.jl:138]: The back ticks aren't necessary. See for example this source code: https://github.com/JuliaLang/julia/blob/5544a0fab7648cfa61fe79cd557a7504a92ec1b5/base/docs/basedocs.jl#L1775-L1779

File: src/SurveyDesign.jl (hunk 1 above)
Comment (iuliadmtru) [reply src/SurveyDesign.jl:94]: Same as before, back ticks aren't necessary (https://github.com/JuliaLang/julia/blob/5544a0fab7648cfa61fe79cd557a7504a92ec1b5/base/docs/basedocs.jl#L1775-L1779).

**Comment by iuliadmtru** [comment]
The continuation of this PR is at #54. I will close this one.

**Pull request closed without merging**